x-pack/plugins/lens/public/xy_visualization/xy_config_panel/threshold_panel.tsx

```diff
@@ -0,0 +1,289 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import './xy_config_panel.scss';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonGroup, EuiComboBox, EuiFormRow, EuiIcon, EuiRange } from '@elastic/eui';
+import type { PaletteRegistry } from 'src/plugins/charts/public';
+import type { VisualizationDimensionEditorProps } from '../../types';
+import { State } from '../types';
+import { FormatFactory } from '../../../common';
+import { YConfig } from '../../../common/expressions';
+import { LineStyle, FillStyle } from '../../../common/expressions/xy_chart';
+
+import { ColorPicker } from './color_picker';
+import { updateLayer, idPrefix } from '.';
+import { useDebouncedValue } from '../../shared_components';
+
+const icons = [
+  {
+    value: 'none',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.noIconLabel', { defaultMessage: 'None' }),
+  },
+  {
+    value: 'asterisk',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.asteriskIconLabel', {
+      defaultMessage: 'Asterisk',
+    }),
+  },
+  {
+    value: 'bell',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.bellIconLabel', {
+      defaultMessage: 'Bell',
+    }),
+  },
+  {
+    value: 'bolt',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.boltIconLabel', {
+      defaultMessage: 'Bolt',
+    }),
+  },
+  {
+    value: 'bug',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.bugIconLabel', {
+      defaultMessage: 'Bug',
+    }),
+  },
+  {
+    value: 'editorComment',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.commentIconLabel', {
+      defaultMessage: 'Comment',
+    }),
+  },
+  {
+    value: 'alert',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.alertIconLabel', {
+      defaultMessage: 'Alert',
+    }),
+  },
+  {
+    value: 'flag',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.flagIconLabel', {
+      defaultMessage: 'Flag',
+    }),
+  },
+  {
+    value: 'tag',
+    label: i18n.translate('xpack.lens.xyChart.thresholds.tagIconLabel', {
+      defaultMessage: 'Tag',
+    }),
+  },
+];
+
+const IconView = (props: { value?: string; label: string }) => {
+  if (!props.value) return null;
+  return (
+    <span>
+      <EuiIcon type={props.value} />
+      {` ${props.label}`}
+    </span>
+  );
+};
+
+const IconSelect = ({
+  value,
+  onChange,
+}: {
+  value?: string;
+  onChange: (newIcon: string) => void;
+}) => {
+  const selectedIcon = icons.find((option) => value === option.value) || icons[0];
+
+  return (
+    <EuiComboBox
+      isClearable={false}
+      options={icons}
+      selectedOptions={[selectedIcon]}
+      onChange={(selection) => {
+        onChange(selection[0].value!);
+      }}
+      singleSelection={{ asPlainText: true }}
+      renderOption={IconView}
+      compressed
+    />
```

Comment:
Is it possible to prepend the selected icon before the icon name in this selector, as shown in the designs? Small issue, so disregard if not possible natively via EUI.

![image](https://user-images.githubusercontent.com/3884767/133114146-b3b897d1-ab06-46cf-bb13-07c67fbe4c35.png)

Reply:
I do not think this is currently possible natively in EUI